jetcd-launcher/src/main/java/io/etcd/jetcd/launcher/EtcdClusterFactory.java

```diff
@@ -35,12 +36,19 @@ public static EtcdCluster buildCluster(@NonNull String clusterName, int nodes, b
   public static EtcdCluster buildCluster(@NonNull String clusterName, int nodes, boolean ssl, boolean restartable) {
     final Network network = Network.builder().id(clusterName).build();
     final CountDownLatch latch = new CountDownLatch(nodes);
+    final AtomicBoolean failedToStart = new AtomicBoolean(false);
     final EtcdContainer.LifecycleListener listener = new EtcdContainer.LifecycleListener() {
       @Override
       public void started(EtcdContainer container) {
         latch.countDown();
       }
 
+      @Override
+      public void failedToStart(EtcdContainer container) {
```

Comment:
you should include the exception in the handler

Reply:
I did that originally: I had a concurrent set where exception from every node was added. Then if set wasn't empty I was collecting `e.getMessage()` and printing `Error while starting cluster:` + failures. Result was: `Error while starting cluster: Failed to start container, Failed to start container, Failed to start container`. It wasn't very helpful.

Given that stack traces are printed nevertheless before I though it would be cleaner just to have a flag if there is a failure.

Could you please share what use case do you have in mind? I was thinking that this class is supposed to be used in test where as I mentioned before stacktrace is printed. But maybe I'm missing some use case where it would be required.
Also what do you think is the best way to deal with multiple exceptions (there could be as many exceptions as there are nodes)?

Reply:
You do not need a set, it could be just `this.listener.failedToStart(this, exception) and log the execption in the failedToStart method then, the stack trace is not more printed because you do not re throw the exception so the original cause is mnissed

Reply:
Got it. Please take a look.